package/base-files/files/bin/board_detect

```diff
@@ -1,14 +1,20 @@
 #!/bin/sh
 
-CFG=$1
+CFG="${1:-/etc/board.json}"
 
-[ -n "$CFG" ] || CFG=/etc/board.json
-
-[ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
-		$(. $a)
+if [ -d '/etc/board.d/' ] && \
+   [ ! -s "${CFG}" ]; then
+	for _script in '/etc/board.d/'*; do
+		if [ ! -s "${_script}" ]; then
+			continue
+		fi
+		# shellcheck source=/dev/null disable=SC2091  # Can't use eval
```

Comment:
What is comment supposed to mean?

Reply:
It's a comment to suppress shellcheck output. In this case: https://www.shellcheck.net/wiki/SC2091

Reply:
Indeed, thank you @neheb  Shellcheck tells us to two things about the following line. First, it sees an 'include' which it can't resolve, so it can't scan it (this if fine, but we have to tell shellcheck this).

Secondly, more importantly, SC2091 is triggered on the fact that we are trying to execute whatever comes out of the subshell, which usually `eval` is recommended to be used. For whatever reason (more on this below) we need this style, and so we tell the reader (and shellcheck) as to why.

Reply:
@jow- @neheb about this, some times ago there was an idea to add a workflow to check each shell script with shellcheck... Wonder if it's acceptable to have script with these comments in to, in the future, have this kind of system.

(crazy idea, on package compile these script are stripped of any comment related to shellcheck)

Reply:
it's probably better to globally disable various checks. Busybox sh is not quite ash nor sh.

Reply:
so we should just disable them globally by adding a shellcheck config file in the root of the repo. Probably a good idea as adding these comments in the shell file seems a bit wrong to me.

(should we also set the default shell to ash anyway? incredible that nobody cared to add to shellcheck a shell type that handle busybox sh)

Reply:
well, yes https://github.com/koalaman/shellcheck/commit/2e59eba6ebcafa029076a37cdea09521860ee01d , but actually no.

AFAIK busybox ash is basically ash + string substitution stuff.

Reply:
Note, that this exception is very specifically for this one situation, we should NOT want to disable SC2091 globally _at all_. Exception to rules.

The =/dev/null you can easily do globally, my sc looks very simple (on purpose)
```
enable=all
external-sources=true
shell=sh
```
Personally, for this specific SC2091, I still want to know, do we have to get the output at all? (I tried, currently we do, for whatever reason) as it does not work otherwise. This should have been really rewritten to work as:
```sh
for _script in ${scripts}; do
 if [ ! -s ${_script:-} ];
    continue
  fi
  ${_script}
done
```
(or better yet, use runparts, which we don't have).

As I wrote, better yet, revert  the commit that dropped executable bits from scripts; why are we sourceing them? What's the purpose? Why can't se not just check for execution bit and actually execute them? What was the reason to not use this 'quite common' pattern?